File: x/pset2/vigenere.c
#include <stdlib.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main( int argc, string argv[] )

{
    if (argc != 2)
    {
        printf ("Usage: ./viginere k\n");
        return 1;
    }
    
    
    string key =  argv[1] ; 
    
    for (int counter = 0; counter < strlen(key); counter++)
    {
        if (isalpha(key[counter]))
        {
            continue;
        }
        
        else
        {
            printf ("Usage: ./viginere k\n");
            return 1;
        }
    }
    
    printf("plaintext: "); 
    string data = get_string();
    printf("ciphertext: ");
    int leng = strlen(key);
    int counter2 = 0; 
    int key2 = 0;
    char symbol;
    
    for (int counter = 0; counter < strlen(data); counter++)
    {
        if (isalpha(data[counter]))   
        {
            counter2 = counter2%leng;
            symbol = key[counter2];
            symbol = tolower(symbol);
            key2 = (symbol - 97)%26;
            
            
              if (islower(data[counter]))    
              {
                  int num_a = ((data[counter] - 97) + key2) %26;
                  num_a = num_a + 97;
                  printf("%c", (char) num_a);
              }
            
            
              else
              {
                  int num_b = ((data[counter] - 65) + key2) %26; 
                  num_b = num_b + 65;
                  printf("%c", (char) num_b);
              }
              counter2++;
        }
        
        else
        {
            printf("%c", data[counter]); //symbols
        }
    }
    
    
    printf("\n");
    
} 

File: x/pset6/vigenere.py
#!/usr/bin/env python3
import sys
import cs50

def main():
    if len(sys.argv) != 2:
        print("Usage: ./vigenere k")
        exit(1)
    else:
        key = sys.argv[1]
        for z in range(len(key)):
            if key.isalpha() == 0:
                print("Usage: ./vigenere k")
                exit(1)
    print("plaintext: ", end="")
    p = cs50.get_string()
    print("ciphertext: ", end="")
    funcvigenere(p, key)
	
def funcvigenere(strl, key):
    counter = 0
    counter2 = 0
    number = 0
    for i in range(len(strl)):
        o = len(key)
        symbol = strl[i]
        if symbol.isalpha():
            counter = counter % o
            number = key[counter].upper()
            number = ord(number)
            counter2 = (number - 65)%26
            if symbol.isupper():
                numb = ord(symbol)
                l = ((numb - 65) + counter2) % 26
                l = l + 65
                print("{}".format(chr(l)), end="")
            if symbol.islower():
                numb = ord(symbol)
                l = ((numb - 97) + counter2) % 26
                l = l + 97
                print("{}".format(chr(l)), end="")
            counter += 1
        else:
            print("{}".format(symbol), end="")
    print("")
        
	
if __name__ == "__main__":
    main()